gc/base/MemorySubSpace.cpp

```diff
@@ -936,8 +936,12 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 		reportSystemGCEnd(env);
 		env->releaseExclusiveVMAccessForGC();
 
+
+		if ((J9MMCONSTANT_EXPLICIT_GC_PREPARE_FOR_CHECKPOINT == gcCode)
 #if defined(OMR_GC_IDLE_HEAP_MANAGER)
-		if ((J9MMCONSTANT_EXPLICIT_GC_IDLE_GC == gcCode) && (_extensions->gcOnIdle)) {
+		|| ((J9MMCONSTANT_EXPLICIT_GC_IDLE_GC == gcCode) && (_extensions->gcOnIdle))
+#endif
+		) {
 			OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
 			uint64_t startTime = omrtime_hires_clock();
 			uintptr_t releasedBytes = _extensions->heap->getDefaultMemorySpace()->releaseFreeMemoryPages(env);
```

Comment:
If you want to use this, you'll have to change #ifdefs from a dozen of files that surround releaseFreeMemoryPages in various classes.
Either completely remove ifdefs or append them with a checkpoint specific build flag.

Reply:
is `OMR_GC_IDLE_HEAP_MANAGER` on by default everywhere? I believe this is true on x86.

Reply:
I could just remove that macro

Reply:
I was thinking about removing ifdefs just from releaseFreeMemoryPages (and dependent APIs), so it's accessible for other higher level functionality like checkpoint. but still keeping them around anything that is very Idle GC specific (control variables like gcOnIdle, various constants, and any code that may use them).